src/coreclr/jit/emit.h

```diff
@@ -2890,6 +2890,7 @@ class emitter
         else
         {
             return (curInsIG->igFlags & IGF_EXTEND) &&
+                   ((curInsIG->igFlags & (IGF_FUNCLET_PROLOG | IGF_FUNCLET_EPILOG)) == 0) &&
```

Comment:
https://github.com/dotnet/runtime/pull/114843 is a better fix, but let's see the CI results first.

Reply:
...and apparently my fix failed to do the job.

Reply:
https://dev.azure.com/dnceng-public/public/_build/results?buildId=1021337&view=logs&j=63f2a02a-7891-5505-e9da-45b13105e16b is green now? 👀

Reply:
The gcstress-extra pipelines still failed on the same tests. I will download the artifacts later on my ARM64 machine to inspect.

Reply:
Maybe it is the different one but this one succeeded https://github.com/dotnet/runtime/runs/40841921822, I was tailing https://helix.dot.net/api/jobs/5d4b8a30-aa43-44dd-a1bc-292fc044b085/workitems?api-version=2019-06-17. 😅

Reply:
This one was the failed one: https://helixr18s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-pull-114630-merge-d4a47512561d457aa6/HardwareIntrinsics_General_ro.3.8/1/console.fe0cd040.log?skoid=8eda00af-b5ec-4be9-b69b-0919a2338892&sktid=72f988bf-86f1-41af-91ab-2d7cd011db47&skt=2025-04-20T15%3A02%3A55Z&ske=2025-04-20T16%3A02%3A55Z&sks=b&skv=2024-11-04&sv=2024-11-04&st=2025-04-20T15%3A02%3A55Z&se=2025-04-20T16%3A02%3A55Z&sr=c&sp=r&sig=rHON%2FF2URccf05vTsCbbVsiwlwiEHF8GkJ4HhMHARE4%3D

The latest run passed it both on win-arm64 and linux-arm64 with @EgorBo's patch.